xbmc/addons/Addon.cpp

```diff
@@ -179,76 +219,79 @@ void CAddon::SaveSettings(void)
 
   // create the XML file
   CXBMCTinyXML doc;
-  if (SettingsToXML(doc))
-    doc.SaveFile(m_userSettingsPath);
+  if (SettingsToXML(instance, doc))
+    doc.SaveFile(data.m_userSettingsPath);
 
-  m_hasUserSettings = true;
+  data.m_hasUserSettings = true;
 
   //push the settings changes to the running addon instance
-  CServiceBroker::GetAddonMgr().ReloadSettings(ID());
+  CServiceBroker::GetAddonMgr().ReloadSettings(ID(), instance);
 #ifdef HAS_PYTHON
   CServiceBroker::GetXBPython().OnSettingsChanged(ID());
```

Comment:
must this as well be extended by the new instance id param?

Reply:
Has ignored it now, but if python becomes the same way a add would be good.